### PR TITLE
light: fix panic with RPC calls to commit and validator when height is nil

### PR DESCRIPTION
**Pull request by cmwaters**
## Description

Closes: #XXX

### CODE REVIEW

File: light/provider/http/http.go [hunk 1]
```diff
@@ -68,7 +68,7 @@ func (p *http) LightBlock(ctx context.Context, height int64) (*types.LightBlock,
 		return nil, err
 	}
 
-	vs, err := p.validatorSet(ctx, h)
+	vs, err := p.validatorSet(ctx, &sh.Height)
```
Comment (melekes) [inline light/provider/http/http.go:71]: how's this better than using `h`?

File: light/rpc/client.go [hunk 2]
```diff
@@ -409,7 +411,16 @@ func (c *Client) BlockResults(ctx context.Context, height *int64) (*ctypes.Resul
...
-	l, err := c.updateLightClientIfNeededTo(ctx, *height)
+	// or at the latest height if no height is provided.
+	var (
+		l   *types.LightBlock
+		err error
+	)
+	if height == nil {
+		l, err = c.lc.Update(ctx, time.Now())
+	} else {
+		l, err = c.updateLightClientIfNeededTo(ctx, *height)
```
Comment (melekes) [inline light/rpc/client.go:422]: should we modify updateLightClientIfNeededTo to accept `nil` height and call Update?

File: light/provider/http/http.go (hunk 1 above)
Comment (cmwaters) [reply light/provider/http/http.go:71]: I'm thinking what happens if h is `nil` i.e. grab the latest height but the height for the latest validator set is different to the height of the latest commit (especially as a commit aka a signed header can only be formed at the following height)

File: light/rpc/client.go (hunk 2 above)
Comment (cmwaters) [reply light/rpc/client.go:422]: yeah I was thinking that. Perhaps that would be better

File: light/provider/http/http.go (hunk 1 above)
Comment (melekes) [reply light/provider/http/http.go:71]: but h can't be nil

Comment (cmwaters) [reply light/provider/http/http.go:71]: really? what about this in `validateHeight`
```
if height == 0 {
	h = nil
}
return h, nil
```

Comment (melekes) [reply light/provider/http/http.go:71]: ah, right.

**Review by melekes — APPROVED** [review]

**Comment by cmwaters** [comment]
I will backport this as well

**Pull request merged**